### PR TITLE
Fixed permissions

### PR DESCRIPTION
**Pull request by HeahDude**
Big PR (better reviewing by commits):
 - 198479d use referer globally (another PR will come to handle event or group registration processes)
 - db7fda4 a simple renaming
 - 448670a improve unregistration perf and clear some missing data, added a CitizenProjectAuthority
 - b1d9f87 refactored voters, allow anonymous registration to all events (including citizen actions) and referents are not granted to create citizen projects anymore.
 - 2b26d2a self explanatory
 - 772125d self explanatory

### CODE REVIEW

File: src/Entity/Adherent.php [hunk 1]
```diff
@@ -281,8 +282,8 @@ public function getRoles(): array
             $roles[] = 'ROLE_BOARD_MEMBER';
         }
 
-        if ($this->isAdministrator()) {
-            $roles[] = 'ROLE_ADMINISTRATOR';
+        if ($this->isProjectAdministrator()) {
+            $roles[] = 'ROLE_PROJECT_ADMINISTRATOR';
```
Comment (thomasbisignani) [inline src/Entity/Adherent.php]: Same here, I think `ROLE_CITIZEN_PROJECT_ADMINISTRATOR` is more precise.

File: src/Entity/Adherent.php [hunk 2]
```diff
@@ -859,7 +862,7 @@ public function isHostOf(Committee $committee): bool
         return $membership->canHostCommittee();
     }
 
-    public function isAdministrator(): bool
+    public function isProjectAdministrator(): bool
```
Comment (thomasbisignani) [inline src/Entity/Adherent.php]: I think `isCitizenProjectAdministrator()` is better and more precise.

File: src/Twig/CitizenProjectExtension.php [hunk 3]
```diff
@@ -11,9 +11,7 @@ public function getFunctions()
     {
         return [
             // Permissions
-            new TwigFunction('is_administrator', [CitizenProjectRuntime::class, 'isAdministrator']),
-            new TwigFunction('is_promotable_administrator', [CitizenProjectRuntime::class, 'isPromotableAdministrator']),
-            new TwigFunction('is_demotable_administrator', [CitizenProjectRuntime::class, 'isDemotableAdministrator']),
+            new TwigFunction('is_project_administrator', [CitizenProjectRuntime::class, 'isProjectAdministrator']),
```
Comment (thomasbisignani) [inline src/Twig/CitizenProjectExtension.php]: `is_citizen_project_administrator`

File: src/Twig/CitizenProjectRuntime.php [hunk 4]
```diff
@@ -4,37 +4,29 @@
...
-        return $this->citizenProjectManager->isPromotableAdministrator($adherent, $citizenProject);
-    }
-
-    public function isDemotableAdministrator(Adherent $adherent, CitizenProject $citizenProject): bool
-    {
-        return $this->citizenProjectManager->isDemotableAdministrator($adherent, $citizenProject);
-    }
-
-    public function isAdministrator(CitizenProject $citizenProject): bool
+    public function isProjectAdministrator(CitizenProject $citizenProject): bool
```
Comment (thomasbisignani) [inline src/Twig/CitizenProjectRuntime.php]: What about using `isAdministratorOf(CitizenProject $citizenProject)` ?

File: src/Security/Voter/Committee/CreateCommitteeVoter.php [hunk 5]
```diff
@@ -2,46 +2,32 @@
...
-        if ($this->manager->isCommitteeHost($adherent)) {
-            return false;
-        }
-
-        if ($this->manager->hasCommitteeInStatus($adherent, CommitteeManager::COMMITTEE_STATUS_NOT_ALLOWED_TO_CREATE_ANOTHER)) {
-            return false;
-        }
-
-        return true;
+        // Cannot create a project when referent or already host one
```
Comment (Remg) [inline src/Security/Voter/Committee/CreateCommitteeVoter.php]: Cannot create a **committee**

File: tests/CitizenProject/CitizenProjectManagerTest.php [hunk 6]
```diff
@@ -171,7 +106,11 @@ protected function setUp()
         ]);
 
         $this->container = $this->getContainer();
-        $this->citizenProjectManager = new CitizenProjectManager($this->getManagerRegistry(), $this->getStorage());
+        $this->citizenProjectManager = new CitizenProjectManager(
+            $this->getManagerRegistry(),
+            $this->getStorage(),
+            $this->createMock(CitizenProjectAuthority::class))
```
Comment (Remg) [inline tests/CitizenProject/CitizenProjectManagerTest.php]:
```
+            $this->createMock(CitizenProjectAuthority::class)
+       )
```

File: src/Security/Voter/Committee/CreateCommitteeVoter.php (hunk 5 above)
Comment (HeahDude) [reply src/Security/Voter/Committee/CreateCommitteeVoter.php]: Copy pasted with the ass :D. Thanks!

File: src/Security/Voter/CitizenProject/AdministrateCitizenProjectVoter.php [hunk 7]
```diff
@@ -2,31 +2,31 @@
...
+class AdministrateCitizenProjectVoter extends AbstractAdherentVoter
 {
-    private $manager;
-
-    public function __construct(CitizenProjectManager $manager)
-    {
-        $this->manager = $manager;
-    }
-
     protected function supports($attribute, $subject): bool
```
[the remark this answers is not in the document]
Comment (HeahDude) [reply src/Security/Voter/CitizenProject/AdministrateCitizenProjectVoter.php]: This is not on me :p but you're right... fixed, thanks!

File: src/Twig/CitizenProjectRuntime.php (hunk 4 above)
Comment (HeahDude) [reply src/Twig/CitizenProjectRuntime.php]: Good idea! Fixed, thx!

File: src/Entity/Adherent.php (hunk 1 above)
Comment (HeahDude) [reply src/Entity/Adherent.php]: Agreed, fixed. Thx!

**Pull request merged**